src/fsharp/TypeChecker.fs

```diff
@@ -8448,7 +8448,7 @@ and TcDelayed cenv overallTy env tpenv mExpr expr exprty (atomicFlag:ExprAtomicF
     match delayed with 
     | []  
     | DelayedDot :: _ -> 
-        UnifyTypesAndRecover cenv env mExpr overallTy exprty
+        UnifyTypes cenv env mExpr overallTy exprty
```

Comment:
I'm not totally comfortable with this fix - I suppose it will do for now but the fix really needs to be in `InferArityOfExpr` or one of the other functions on the stack shown here https://github.com/Microsoft/visualfsharp/issues/5468#issuecomment-411775861

Basically, the use of `UnifyTypesAndRecover` should always be safe.  Later procedures that operate on the TAST need to be robust in the face of some mismatches in inferred types - in this case the mismtach is between the r.h.s. of a definition be a lambda but the overall type is not a function type.  The procedure that is assuming that things are good is InferArityOfExpr and so that's where the fix really should be.  

I'm ok with this fix for now though

Reply:
Gotcha. If the goal is to be able to recover safely at the point in the time, then something needs to be done elsewhere. I don't think the fix can be in `InferArityOfExpr`, it will need to be on one of the other functions.

We still have a bit of time. I can look at another possible fix without touching this. I do want to do a more correct fix if possible.

Reply:
After looking through the functions noted in the stack trace, I do not know a good way to handle this other than the fix presented here.

For some history, everything used to be `UnifyTypes`. There was no recovery. When recovery was added, it was added in two places. Reverting one place fixes our internal error and doesn't regress the completions fix; this feels like the right thing to do.

If we are going to add recovery back here, we need an example of what this affects; so far, I haven't found anything.